src/main/java/org/truffleruby/core/numeric/FloatNodes.java

```diff
@@ -883,59 +882,61 @@ protected double toF(double value) {
     public abstract static class ToSNode extends CoreMethodArrayArgumentsNode {
 
         @Child private StringNodes.MakeStringNode makeStringNode = StringNodes.MakeStringNode.create();
+        @Child protected TruffleString.FromJavaStringNode fromJavaStringNode = TruffleString.FromJavaStringNode
+                .create();
 
         /* Ruby has complex custom formatting logic for floats. Our logic meets the specs but we suspect it's possibly
          * still not entirely correct. JRuby seems to be correct, but their logic is tied up in their printf
          * implementation. Also see our FormatFloatNode, which I suspect is also deficient or under-tested. */
 
         @Specialization(guards = "value == POSITIVE_INFINITY")
         protected RubyString toSPositiveInfinity(double value,
-                @Cached("specialValueRope(POSITIVE_INFINITY)") Rope cachedRope) {
-            return createString(cachedRope, Encodings.US_ASCII);
+                @Cached("specialValueString(POSITIVE_INFINITY, fromJavaStringNode)") TruffleString cachedString) {
+            return createString(cachedString, Encodings.US_ASCII);
         }
 
         @Specialization(guards = "value == NEGATIVE_INFINITY")
         protected RubyString toSNegativeInfinity(double value,
-                @Cached("specialValueRope(NEGATIVE_INFINITY)") Rope cachedRope) {
-            return createString(cachedRope, Encodings.US_ASCII);
+                @Cached("specialValueString(NEGATIVE_INFINITY, fromJavaStringNode)") TruffleString cachedString) {
+            return createString(cachedString, Encodings.US_ASCII);
         }
 
         @Specialization(guards = "isNaN(value)")
         protected RubyString toSNaN(double value,
-                @Cached("specialValueRope(value)") Rope cachedRope) {
-            return createString(cachedRope, Encodings.US_ASCII);
+                @Cached("specialValueString(value, fromJavaStringNode)") TruffleString cachedString) {
+            return createString(cachedString, Encodings.US_ASCII);
         }
 
         @Specialization(guards = "hasNoExp(value)")
         protected RubyString toSNoExp(double value) {
-            return makeStringNode.executeMake(makeRopeNoExp(value, getLanguage().getCurrentThread()),
-                    Encodings.US_ASCII, CodeRange.CR_7BIT);
+            return makeStringNode.executeMake(makeStringNoExp(value, getLanguage().getCurrentThread()),
+                    Encodings.US_ASCII, CodeRange.ASCII);
```

Comment:
What I've been doing is importing `ASCII` statically, to not have the potential problem of having two different `CodeRange` imported.
But it seems fine in this file as there is no conflict.
Might be good to do static imports everywhere for consistency, unsure.